addons/vscode/src/extension.ts

```diff
@@ -493,7 +495,7 @@ const launchPreview = async (task: LaunchInBrowserTask | LaunchInWebViewTask) =>
 		html = html.replace(
 			"preview-arg:previewMode:Doc",
 			`preview-arg:previewMode:${previewMode}`
-		);
+		).replace("preview-arg:state:{}", `preview-arg:state:${JSON.stringify({ mode: task.mode, fsPath: bindDocument.uri.fsPath.replace(/`/g, "").replace(/"/g, "") })}`);
```

Comment:
It seems code here removes backticks and quote from `fsPath`, Could you add comment here to explain why it is needed?

Reply:
Added a comment, also, I think we only need to replace the backticks. This is to make backticks don't cause the string's end in the JS itself (since this solution is fragile)

Reply:
one trick is that we can use base64 to encode the payload of the content so there is no special characters. And the browser/webview side can perform base64 decode first then `JSON.parse`

Reply:
Yeah, that seems like a good idea. Isn't escaping the backticks sufficient though?

Reply:
Nevermind, backslashes also cause problems. And dollar signs... sigh.

Reply:
Ok, switched to Base64 now so that should be good